test/e2e/framework/util.go

```diff
@@ -402,6 +403,19 @@ func SkipIfContainerRuntimeIs(runtimes ...string) {
 	}
 }
 
+func SkipUnlessDefaultStorageClassPresent(c clientset.Interface) {
+	list, err := c.StorageV1().StorageClasses().List(metav1.ListOptions{})
+	if err != nil {
+		Skipf("Error listing storage classes: %v", err)
+	}
+	for _, sc := range list.Items {
+		if storageutil.IsDefaultAnnotation(sc.ObjectMeta) {
+			return
+		}
+	}
+	Skipf("No default storage class found")
```

Comment:
Can we record the result so we don't have to visit the API every time?

Reply:
Addressed in latest PR. Also provided an option to refresh if caching leads to instability.

Reply:
IMO it's safer to query every time rather than relying on a nearly global internal state. The performance hit per query isn't significant, not a lot of tests use this function today, and we avoid the potential error that a default storage class is only missing *temporarily*